src/components/UNSTABLE__Pagination/PageInput.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * @file Page input.
+ * @copyright IBM Security 2020
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import setupGetInstanceId from 'carbon-components-react/lib/tools/setupGetInstanceId';
+import NumberInput from '../NumberInput';
+import { appendComponentNamespace } from '../../globals/namespace/index';
+import { namespace as paginationNamespace } from './Pagination';
+
+const namespace = appendComponentNamespace(paginationNamespace, 'page-input');
+const getInstanceId = setupGetInstanceId();
+
+function PageInput({
+  className,
+  currentPage,
+  id,
+  invalidText,
+  label,
+  max,
+  min,
+  totalPages,
+  ...other
+}) {
+  const instanceId = `${namespace}__input-${getInstanceId()}`;
+
+  return (
+    <NumberInput
+      className={classnames(namespace, className)}
+      hideLabel
+      id={instanceId || id}
+      invalidText={invalidText}
+      label={label}
+      value={currentPage}
+      max={max || totalPages}
+      min={min}
```

Comment:
Even though ideally someone wouldn't be setting these values, seems worthwhile to let them override defaults. 🤷‍♀